Challenges/StacksQueues/AnimalShelter.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using DataStructures.StacksQueues;
+using System.Collections;
+
+
+namespace Challenges.StacksQueues
+{
+    public class AnimalShelter<T> where T : Animal
+    {
+        private Queue<T> Animals = new Queue<T>();
+
+        public void Enqueue(T animal)
+        {
+            Animals.EnQueue(animal);
+        }
+
+
+
+        public T DeQueue(AnimalType type)
+        {
+          
+            Node<T> result = new Node<T>();
+            Queue<T> temp = new Queue<T>();
+            switch (type)
+            {
+                case AnimalType.Cat:
+                    
+                    while (Animals.isEmpty() == false)
+                    {
+                        T front = Animals.Peek();
+                        if (front.Type == AnimalType.Cat)
```

Comment:
I think you can drop the `switch` altogether if you use `front.Type == type` here.